defaults/serverConfig/zowe.yaml

```diff
@@ -0,0 +1,54 @@
+components:
+  app-server:
+    node:
+      https:
+        ipAddresses:
+        - 0.0.0.0
+        port: 7556
+        # pfx (string), keys, certificates, certificateAuthorities, and certificateRevocationLists are all valid here.
+        keys:
+        - "../defaults/serverConfig/zlux.keystore.key"
+        certificates:
+        - "../defaults/serverConfig/zlux.keystore.cer"
+        certificateAuthorities:
+        - "../defaults/serverConfig/apiml-localca.cer"
+      mediationLayer:
+        server:
+          hostname: localhost
+          port: 10011
+          gatewayPort: 10012
+          isHttps: true
+        enabled: false
+      headers:
+        X-frame-Options:
+          override: true
+          value: sameorigin
+    # All paths relative to zlux-app-server/bin
+    # In real installations, these values will be configured during the install.
+    productDir: "../defaults"
+    siteDir: "../deploy/site"
+    instanceDir: "../deploy/instance"
+    groupsDir: "../deploy/instance/groups"
+    usersDir: "../deploy/instance/users"
+    pluginsDir: "../defaults/plugins"
```

Comment:
your convert-env.sh enforces that the value of these at runtime is an offset of workspace directory.
currently env overrides config.
so, there's no way these would ever be used.

i think it is best to remove these from the defaults zowe.yaml, and update convert-env.sh to do:
```
if [ -z "$ZWE_components_app_server_siteDir" ]; then
  export ZWED_siteDir="$DESTINATION/site"
fi
```
this way people could do overrides in the production zowe.yaml, but if it is not defined, then we'd apply a workspace-based default.

these lines in the default zowe.yaml then could be commented out as
```
#    productDir: "../defaults"
#    siteDir: "~/.zowe/workspace/app-server/site"
#   instanceDir: "~/.zowe/workspace/app-server"
#    groupsDir: "~/.zowe/workspace/app-server/groups"
#    usersDir: "~/.zowe/workspace/app-server/users"
#   pluginsDir: "~/.zowe/workspace/app-server/plugins"
```